cognite/neat/rules/exporter/_rules2dms.py

```diff
@@ -521,56 +498,186 @@ def find_existing_views(self, client: CogniteClient) -> set:
 
-    def create_containers(self, client: CogniteClient):
+    def create_containers(self, client: CogniteClient, existing_containers: set | None = None):
         for container_id in self.containers:
-            logging.info(f"Creating container {container_id} in space {self.space}")
-            _ = client.data_modeling.containers.apply(self.containers[container_id])
+            if not existing_containers or container_id not in existing_containers:
+                logging.info(f"Creating container {container_id}")
+                _ = client.data_modeling.containers.apply(self.containers[container_id])
+            else:
+                warnings.warn(
+                    f"Container {container_id} already exists! Skipping !!!",
+                    stacklevel=2,
+                )
+                logging.info(f"Container {container_id} already exists! Skipping !!!")
 
-    def create_views(self, client: CogniteClient):
+    def create_views(self, client: CogniteClient, existing_views: set | None = None):
         for view_id in self.views:
-            logging.info(f"Creating view {view_id} version {self.views[view_id].version} in space {self.space}")
-            _ = client.data_modeling.views.apply(self.views[view_id])
-
-    def create_data_model(self, client: CogniteClient):
-        logging.info(f"Creating data model {self.external_id} version {self.version} in space {self.space}")
-        _ = client.data_modeling.data_models.apply(
-            DataModelApply(
-                name=self.name,
-                description=self.description,
-                space=self.space,
-                external_id=self.external_id,
-                version=self.version,
-                views=list(self.views.values()),
+            if not existing_views or view_id not in existing_views:
+                logging.info(f"Creating view {view_id}")
+                _ = client.data_modeling.views.apply(self.views[view_id])
+            else:
+                warnings.warn(
+                    f"View {view_id}/{self.views[view_id].version} already exists! Skipping !!!",
+                    stacklevel=2,
+                )
+                logging.info(f"View {view_id}/{self.views[view_id].version} already exists! Skipping !!!")
+
+    def create_data_model(
+        self,
+        client: CogniteClient,
+        existing_data_model: DataModelId | None = None,
+    ):
+        if not existing_data_model:
+            logging.info(f"Creating data model {self.external_id} version {self.version} in space {self.space}")
+            _ = client.data_modeling.data_models.apply(
+                DataModelApply(
+                    name=self.name,
+                    description=self.description,
+                    space=self.space,
+                    external_id=self.external_id,
+                    version=self.version,
+                    views=list(self.views.values()),
```

Comment:
Here you are recreating the views. Is that intentional? What about just referencing them?
```suggestion
                    views=[view.as_id() for view in self.views],
```

Reply:
good catch ! need only ids!

Reply:
fixed!